schema/schema_validator.py

```diff
@@ -369,18 +384,24 @@ def check_schema_files(self):
 
         return schema_errors
 
-    def validate_test_output_parallel(self):
+    def validate_all_test_output(self):
         test_validation_plans = self.get_test_validation_plans()
-        num_processors = mp.cpu_count()
-        logging.info('JSON test output validation: %s processors for %s plans', num_processors,
-                     len(test_validation_plans))
+        if not self.run_serial:
+            num_processors = mp.cpu_count()
+            logging.info('JSON test output parallel validation: %s processors for %s plans', num_processors,
+                         len(test_validation_plans))
 
-        # How to get all the results
-        processor_pool = mp.Pool(num_processors)
-        with processor_pool as p:
-            results = p.map(self.validate_json_file, test_validation_plans)
+            # How to get all the results
+            processor_pool = mp.Pool(num_processors)
+            with processor_pool as p:
+                results = p.map(self.validate_json_file, test_validation_plans)
 
-        return results, test_validation_plans
+            return results, test_validation_plans
+        else:
+            logging.info('JSON test output serially validation on %s files!', len(test_validation_plans))
+            for test_data in test_validation_plans:
```

Comment:
ditto. also, does `results` need to be initialized before you call `.append` on it?